icubam/predicu/data.py

```diff
@@ -44,58 +76,58 @@
 }
 
 
-def load_all_data(
+def load_if_not_cached(data_source, cached_data, **kwargs):
+  if data_source not in DATA_SOURCES:
+    raise ValueError(f"Unknown data source: {data_source}")
+  load_data_fun = globals().get(f"load_{data_source}", None)
+  if load_data_fun is None:
+    raise RuntimeError(
+      f"No loading function associated to data source {data_source}"
+    )
+  if cached_data is None or data_source not in cached_data:
+    return load_data_fun(cached_data=cached_data, **kwargs)
+  return cached_data[data_source]
+
+
+def load_bedcounts(
   clean=True,
   spread_cum_jump_correction=False,
   api_key=None,
+  icubam_host=None,
   max_date=None,
-  icubam_data: pd.DataFrame = None,
-  load_pre_icubam_data: bool = False,
+  cached_data=None,
 ):
-  icubam = load_icubam_data(data=icubam_data, api_key=api_key)
-  if load_pre_icubam_data:
-    pre_icubam = load_pre_icubam_data()
-    dates_in_both = set(icubam.date.unique()) & set(pre_icubam.date.unique())
-    pre_icubam = pre_icubam.loc[~pre_icubam.date.isin(dates_in_both)]
-    d = pd.concat([pre_icubam, icubam])
-  else:
-    d = icubam
+  d = load_if_not_cached(
+    "icubam", cached_data, api_key=api_key, icubam_host=icubam_host
+  )
   if clean:
     d = clean_data(d, spread_cum_jump_correction)
   d = d.sort_values(by=["date", "icu_name"])
   if max_date is not None:
     logging.info("data loaded's max date will be %s (excluded)" % max_date)
     d = d.loc[d.date < pd.to_datetime(max_date).date()]
+  d = d.loc[d.department.isin(DEPARTMENTS_GRAND_EST)]
```

Comment:
I find this strange. Nothing in the function name or description suggests that discards data for departments other than Grand Est.  Are sure about it?

I see it was done in plots earlier. If you want to avoid repetition, maybe, pass an optional `deparment=None` (i.e. no filtering by default) to `load_bedcounts` and then use load_bedcounts(department="Grand-Est") in all plots. The danger of current factorization is that from plots it's very hard to know this filtering is happening.

Reply:
the original analysis was on the Grand Est region and any ICU that was not in the Grand Est region was discarded from the analysis. this should be removed, i agree.